src/autoapp/Service/ServiceFactory.cpp

```diff
@@ -58,7 +59,14 @@ ServiceFactory::ServiceFactory(boost::asio::io_service& ioService, configuration
 #ifdef USE_OMX
     , omxVideoOutput_(std::make_shared<projection::OMXVideoOutput>(configuration_, this->QRectToDestRect(screenGeometry_), activeCallback_))
 #endif
+    , nightMode_(nightMode)
 {
+    #ifdef USE_GST
```

Comment:
can you do this in the initialization list with the `#elif defined` (and just use the comma operator to call the QGst init function)